x-pack/test/security_solution_api_integration/test_suites/entity_analytics/utils/entity_store.ts

```diff
@@ -68,6 +70,54 @@ export const EntityStoreUtils = (
     expect(res.status).to.eql(200);
   };
 
+  const initEntityEngineForEntityTypeAndWait = async (entityType: EntityType) => {
+    await initEntityEngineForEntityType(entityType);
+
+    let attempts = 10;
+    let lastBody: any = null;
+    const delayMs = 2000;
+
+    while (attempts > 0) {
+      const { body } = await api.getEntityEngine({ params: { entityType } }, namespace).expect(200);
+      lastBody = body;
+      if (body.status === 'started') {
+        return;
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, delayMs));
+      attempts--;
+    }
+
+    throw new Error(
+      `Engine for entity type ${entityType} did not start after multiple attempts, last status: ${JSON.stringify(
+        lastBody
+      )}`
+    );
+  };
+
+  const initEntityEngineForEntityTypesAndWait = async (entityTypes: EntityType[]) => {
+    await Promise.all(entityTypes.map((entityType) => initEntityEngineForEntityType(entityType)));
+
+    let attempts = 10;
+    let lastBody: any = null;
+    const delayMs = 2000;
+
+    while (attempts > 0) {
```

Comment:
There is a utility for this here: https://github.com/elastic/kibana/blob/f4f71d11307276f8eafcfbb8b3f872f60d03853b/x-pack/test/common/utils/security_solution/detections_response/wait_for.ts#L11

And there is also a `retry` service available but I've never used it. `const retry = getService('retry');`